File: leetcode-jvm/src/main/java/com/lone/leetcode/j/Solution14.java
package com.lone.leetcode.j;

/**
 * https://leetcode.com/problems/longest-common-prefix/
 * @author lone
 * @date 2018/11/14
 */
public class Solution14 {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length < 1) {
            return "";
        }
        if (strs.length == 1) {
            return strs[0];
        }
        String result = "";
        int i = 0;
        while (i < strs[0].length()) {
            char c = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if ( i >= strs[j].length() || c != strs[j].charAt(i)) {
                    return result;
                }
            }
            result += c;
            i++;
        }
        return result;
    }
}

File: leetcode-jvm/src/main/kotlin/com/lone/leetcode/k/Solution14.kt
package com.lone.leetcode.k

/**
 * https://leetcode.com/problems/longest-common-prefix/
 * @author lone
 * @date  2018/11/14
 */
class Solution14 {

    fun longestCommonPrefix(strs: Array<String>): String {
        if (strs.size == 0) {
            return ""
        }
        if (strs.size == 1) {
            return strs[0]
        }
        var ret = ""
        var idx = 0
        while (idx < strs[0].length) {
            val cur = strs[0][idx]
            for( start in 1 until  strs.size ) {
                if (idx >= strs[start].length || strs[start][idx] != cur) {
                    return ret
                }
            }
            ret += cur
            idx++
        }
        return ret
    }
}